### ISSUE TITLE
request help: i want to rewrite the response and then return to the client

### ISSUE THREAD

**Issue by rongxianjie** [apache/apisix-java-plugin-runner#232]
### Issue description
I want to get the upstream response content in the plugin for processing and then return to the client, but it return null
This is how the class is handled：
@Override
    public void postFilter(PostRequest request, PostResponse response, PluginFilterChain chain) {
        log.warn("ClientAuthenticationFilter is running");
       
        log.warn("status:{}", request.getUpstreamStatusCode());
       // get the upstream response body
        String responseBody = request.getBody();
        ResultData<ActivityLoginUser> resultData = JsonUtils.json2Bean(responseBody , ResultData.class);
        ResultData<AuthUserVo> authResult = new ResultData<>();
        authResult.setCode(resultData.getCode());
        authResult.setMessage(resultData.getMessage());
        if(ObjectUtils.isNotEmpty(resultData.getData())){
            AuthUserVo authUserVo = processAuthResult(resultData.getData());
            authResult.setData(authUserVo);
        }
        response.setBody(JsonUtils.bean2Json(authResult));
         chain.postFilter(request, response);
    }

java.lang.NullPointerException: null
at com.taikang.tlife.activity.apisix.plugins.fliters.ClientAuthenticationFilter.postFilter(ClientAuthenticationFilter.java:85) 
### Environment

* your apisix-java-plugin-runner version 0.4.0

**Comment by tzssangglass** [comment]
> com.taikang.tlife.activity.apisix.plugins.fliters.ClientAuthenticationFilter.postFilter(ClientAuthenticationFilter.java:85)

Is this the NPE in your private code?

**Comment by rongxianjie** [comment]
yes, it the plugin of the authentication,  the implement fo the plugin is like below:

@Component
public class ClientAuthenticationFilter implements PluginFilter {

    @Override
    public void postFilter(PostRequest request, PostResponse response, PluginFilterChain chain) {
        log.warn("ClientAuthenticationFilter is running");
        log.warn("status:{}", request.getUpstreamStatusCode());

        **//i want to get the upstream body, but it cant ,  the reponseBody is null, so it throws the NullPointerException**
        String responseBody = request.getBody();
       
        String authResult  = processAuthResult(responseBody );

        response.setBody(authResult  );
        chain.postFilter(request, response);
    }

**Comment by rongxianjie** [comment]
yes, it the plugin of the authentication,  the implement fo the plugin is like below:

@Component
public class ClientAuthenticationFilter implements PluginFilter {

    @Override
    public void postFilter(PostRequest request, PostResponse response, PluginFilterChain chain) {
        log.warn("ClientAuthenticationFilter is running");
        log.warn("status:{}", request.getUpstreamStatusCode());

        **//i want to get the upstream body, but it cant ,  the reponseBody is null, so it throws the NullPointerException**
        String responseBody = request.getBody();
       
        String authResult  = processAuthResult(responseBody );

        response.setBody(authResult  );
        chain.postFilter(request, response);
    }

**Comment by tzssangglass** [comment]
whihe line cause the NPE? You need to give detailed steps for reproduction.

**Comment by tzssangglass** [comment]
> the responseBody is null, when process it in processAuthResult, it throws NullPointerException.

You need to give detailed steps for reproduction.

**Comment by tzssangglass** [comment]
a good reproduct describtion like: https://github.com/apache/apisix-java-plugin-runner/issues/189#issuecomment-1226756930

**Comment by gaoxingliang** [comment]
do you add those lines?
```
    @Override
    public List<String> requiredVars() {
        return null;
    }

    @Override
    public Boolean requiredBody() {
        return null;
    }
```
